fabric-dimensions-v1/src/main/java/net/fabricmc/fabric/api/dimension/FabricDimensionType.java

```diff
@@ -0,0 +1,89 @@
+package net.fabricmc.fabric.api.dimension;
+
+import net.minecraft.entity.Entity;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+import net.minecraft.world.World;
+import net.minecraft.world.dimension.Dimension;
+import net.minecraft.world.dimension.DimensionType;
+
+import java.util.Objects;
+import java.util.function.BiFunction;
+
+/**
+ * Fabric version of DimensionType.
+ * DimensionType is a registry wrapper for Dimension.
+ * Stores a default {@link EntityPlacer} which is used to place entities in the world when they join.
+ * Instances of this class get their raw ids automatically assigned.
+ */
+public class FabricDimensionType extends DimensionType {
+    private final Identifier identifier;
+    private final EntityPlacer defaultPlacement;
+    private int fixedId;
+
+    /**
+     * @param name          a unique name used both for the dimension suffix and the save directory
+     * @param factory       a function creating new {@code Dimension} instances
+     * @param defaultPlacer a default {@code EntityPlacer} for the dimension
+     * @param hasSkyLight   {@code true} if the dimension type should have skylight like the overworld,
+     *                      {@code false} otherwise
+     * @throws NullPointerException if any of the parameters is null
+     */
+    public FabricDimensionType(Identifier name, BiFunction<World, DimensionType, ? extends Dimension> factory, EntityPlacer defaultPlacer, boolean hasSkyLight) {
+        // Pass an arbitrary raw id that does not map to any vanilla dimension. That id should never get used.
+        super(3, name.getNamespace() + "_" + name.getPath(), "DIM_" + name.getNamespace() + "_" + name.getPath(), factory, hasSkyLight);
+        Objects.requireNonNull(factory);
+        this.identifier = name;
```

Comment:
It's a unique name, not necessarily the one used to register it. I reckon mods may want to retrieve it, as it is used to compute most names and paths related to the dimension.

Reply:
I don't like the idea of having two different names for the same thing, even if the uses are not quite identical. Since you appear to need it earlier than vanilla, a direct register method and some verification in the vanilla register method should help?

Reply:
Are you suggesting adding a mixin to `Registry.register` to check that a FabricDimensionType's registry id and unique name match? That seems like unnecessary maintenance cost. A static helper method that constructs and register a dimension with the same ID could be useful though.

Reply:
I was thinking more about `net.minecraft.world.dimension.DimensionType.register(String, DimensionType)`, quite annoying this can't be easily & fully covered..

Reply:
That method is private, I don't think anyone would use it.

Reply:
Right, I overlooked that. So just the static register helper then.